### PR TITLE
[openwebnet] Add a time stamp when an alarm zone event occurs

### PR DESCRIPTION
**Pull request by fabgio**
This PR allows to show a time stamp when an alarm zone event occurs on Bticino/Legrand Alarm systems such as an Intrusion or a Tampering
This PR closes https://github.com/openhab/openhab-addons/issues/14535

### <b>Description</b>

This PR allows to show a time stamp when an alarm zone event occurs, i.e. MM/DD/YY,HH:MM INTRUSION

### <b>Test</b>
Tested on BTicino/Legrand bulgrar-alarm unit 3486

### CODE REVIEW

**Comment by fabgio** [comment]
@mvalla I did a commit which followed your  comment but as you see I reverted it as the integrations' tests   had regressions.  So I propose this solution that has been tested successfully on my plant...
The channel ALARM_ZONE_ALARM correctly shows the time stamp and the associaciated event as well. Moreover,  It correctly  resets to None  when the system is re-armed...
LocalDateTime class is also thread safe (Javadoc 17.0) to ensure that no race conditions would heppen (I was worry about)....

**Comment by fabgio** [comment]
@mvalla How should I act for the binding to be tested? I pubblished an announcement on market place but i have been told that  only one contributor per binding is authorized to pubblish. and to contact you to coordinate us

File: bundles/org.openhab.binding.openwebnet/README.md [hunk 1]
```diff
@@ -221,13 +221,13 @@ OPEN command to execute: *5*8#134##
...
-| `armed`                      | `bus_alarm_system`                     | Switch      | Alarm system is armed (`ON`) or disarmed (`OFF`)                      |      R      |
-| `alarm`                      | `bus_alarm_zone`                       | String      | Current alarm for the zone  (`SILENT`, `INTRUSION`, `TAMPERING`, `ANTI_PANIC`) |      R      |
+| Channel Type ID (channel ID) | Applies to Thing Type IDs              | Item Type   | Description                                                                                                                                  | Read/Write  |
+|------------------------------|----------------------------------------|-------------|----------------------------------------------------------------------------------------------------------------------------------------------|:-----------:|
+| `state`                      | `bus_alarm_system`, `bus_alarm_zone`   | Switch      | Alarm system or zone is active (`ON`) or inactive (`OFF`)                                                                                    |      R      |
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/README.md:226]: do not modify other lines if they are not related to the PR

File: bundles/org.openhab.binding.openwebnet/README.md [hunk 2]
```diff
@@ -221,13 +221,13 @@ OPEN command to execute: *5*8#134##
...
+| `network`                    | `bus_alarm_system`                     | Switch      | Alarm system network state (`ON` = network ok, `OFF` = no network)                                                                           |      R      |
+| `battery`                    | `bus_alarm_system`                     | String      | Alarm system battery state (`OK`, `FAULT`, `UNLOADED`)                                                                                       |      R      |
+| `armed`                      | `bus_alarm_system`                     | Switch      | Alarm system is armed (`ON`) or disarmed (`OFF`)                                                                                             |      R      |
+| `alarm`                      | `bus_alarm_zone`                       | String      | Current alarm for the zone  (`SILENT`, `INTRUSION`, `TAMPERING`, `ANTI_PANIC`) as well as the date and time of the event (YY/MM/DD hh:mm:ss) |      R      |
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/README.md:230]: this channel should remain as it was before: it's not a good idea to put together two differnet information: the type of alarm that was triggered an its timestamp. This is not a good desing at all, as it makes difficult for example for scripts to detect which type of alarm has been triggered (you would need to parse the string). 
As already suggested, have a look to the Verisure Alarm binding, where there is one channel for the alarm type (of type String) and one for the timestamp (of type DateTime)

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java [hunk 3]
```diff
@@ -229,27 +233,29 @@ private void updateZoneState(WhatAlarm w) {
     }
 
     private void updateZoneAlarm(WhatAlarm w) {
+        LocalDateTime now = LocalDateTime.now(ZoneId.systemDefault());
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("MM-dd-yyyy HH:mm:ss");
         switch (w) {
             case ZONE_ALARM_INTRUSION:
-                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_INTRUSION));
+                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_INTRUSION + " " + now.format(format)));
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java:240]: timestamp should not be concatenated to the `alarm` channel, but returned in a new `timestamp` channel. This is not a good design. See explanation in previous comment

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java [hunk 4]
```diff
@@ -229,27 +233,29 @@ private void updateZoneState(WhatAlarm w) {
...
                 break;
             case ZONE_ALARM_TAMPERING:
-                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TAMPERING));
+                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TAMPERING + " " + now.format(format)));
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java:243]: see previous comment

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java [hunk 5]
```diff
@@ -229,27 +233,29 @@ private void updateZoneState(WhatAlarm w) {
...
                 break;
             case ZONE_ALARM_ANTI_PANIC:
-                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_ANTI_PANIC));
+                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_ANTI_PANIC + " " + now.format(format)));
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java:246]: see previous comment

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java [hunk 6]
```diff
@@ -229,27 +233,29 @@ private void updateZoneState(WhatAlarm w) {
...
                 break;
             case ZONE_ALARM_SILENT:
-                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_SILENT));
+                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_SILENT + " " + now.format(format)));
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java:249]: see previous comment

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java [hunk 7]
```diff
@@ -229,27 +233,29 @@ private void updateZoneState(WhatAlarm w) {
...
                 break;
             case ZONE_ALARM_TECHNICAL:
-                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TECHNICAL));
+                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TECHNICAL + " " + now.format(format)));
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java:252]: see previous comment

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java [hunk 8]
```diff
@@ -229,27 +233,29 @@ private void updateZoneState(WhatAlarm w) {
...
                 break;
             case ZONE_ALARM_TECHNICAL_RESET:
-                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TECHNICAL_RESET));
+                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_TECHNICAL_RESET + " " + now.format(format)));
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java:255]: see previous comment

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java [hunk 9]
```diff
@@ -229,27 +233,29 @@ private void updateZoneState(WhatAlarm w) {
...
                 break;
             default:
-                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_NONE));
+                updateState(CHANNEL_ALARM_ZONE_ALARM, new StringType(ALARM_NONE + " " + now.format(format)));
```
Comment (mvalla) [inline bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java:258]: see previous comment

**Review by mvalla — CHANGES_REQUESTED** [review]
@fabgio see my review.
I think current solution has a wrong design since you are putting together two information in the same channel. This solution also generates an unwanted breaking change because you are re-defining the alarm channel.
Other comments:
- README.md neeeds will need to be updated with the new `timestamp` channel documentation
- the new `timestamp` channel should be added to the .xml definitions (channels.xml and BusAlarmZone.xml)
- you also have to generate the default English localization, see: https://github.com/openhab/openhab-addons#translations

File: bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java (hunk 9 above)
Comment (fabgio) [reply bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAlarmHandler.java:258]: I see, ok  I will keep working from the base you suggested. Do not consider this PR!
